src/iotHubModuleExplorer.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+"use strict";
+import * as iothub from "azure-iothub";
+import * as vscode from "vscode";
+import { BaseExplorer } from "./baseExplorer";
+import { Constants } from "./constants";
+import { DeviceItem } from "./Model/DeviceItem";
+import { ModuleItem } from "./Model/ModuleItem";
+import { TelemetryClient } from "./telemetryClient";
+import { Utility } from "./utility";
+
+export class IotHubModuleExplorer extends BaseExplorer {
+    constructor(outputChannel: vscode.OutputChannel) {
+        super(outputChannel);
+    }
+
+    public async createModule(deviceItem?: DeviceItem) {
+        const label = "Module";
+        deviceItem = await Utility.getInputDevice(deviceItem, Constants.IoTHubAICreateModuleStartEvent);
+        if (!deviceItem) {
+            return;
+        }
+
+        const iotHubConnectionString = await Utility.getConnectionString(Constants.IotHubConnectionStringKey, Constants.IotHubConnectionStringTitle);

```

Comment:
Do we need hub string to create module? I think device connection string should be enough

Reply:
It is needed, It is in service SDK: `const registry: iothub.Registry = iothub.Registry.fromConnectionString(iotHubConnectionString);`